src/utils/components/SidebarEditor/utils.ts

```diff
@@ -8,3 +8,65 @@ export const safeLoad = <Resource>(value: string): Resource | undefined => {
     return;
   }
 };
+
+export type LineRange = { start: number; end: number };
+
+const getLineFromPath = (resourceYAML: string, path): LineRange => {
+  const yamlLines = resourceYAML.split('\n');
+  let startRange = 0;
+  let endRange = yamlLines.length - 1;
+
+  const properties = path.split('.');
+
+  for (const propertyDepth in properties) {
+    const property = properties[propertyDepth];
+
+    // at every iteration, go one level deeper, remove initial indentation for that range.
+    const replaceIndentationRegex = new RegExp(`^[ ]{${2 * parseInt(propertyDepth)}}`);
+
+    const rangeLines = yamlLines
+      .slice(startRange + 1, endRange)
+      .map((line) => line.replace(replaceIndentationRegex, ''));
+
+    // find the property
+    const startPropertyRange = rangeLines.findIndex((line) => line.startsWith(`${property}:`));
+
+    // find next property at same depth level
+    let endPropertyRange = rangeLines.findIndex(
+      (line, index) => index > startPropertyRange && line.match(/^[A-z]+:/g),
+    );
+
+    if (endPropertyRange === -1) endPropertyRange = rangeLines.length;
```

Comment:
can be removed?

Reply:
Considering the previous suggestion i can do something different but in general i need this.
Here i understand that the range is finished because i find another property using the regex `/^[A-z]+:/g`.

So for example if i want to select the property `spec` i know the start because it starts with `spec:` and i know the end because i encounter another property at the same level `status:` with his conditions.

In that case i know that the spec range is finished. 

But if I want to select the property status, i know that the range is finished because there isn't another property after. So here `endPropertyRange` would be -1. the `findIndex` function does not find anything else after.

So at that point, i know that the end is equivalent to the last line

Reply:
Ok cool. But what's wrong in the code I suggested?

Reply:
Oh I see you took part of it